src/devices/cpu/tlcs900/tmp94c241.cpp

```diff
@@ -0,0 +1,1768 @@
+}
+
+void tmp94c241_device::treg3_w(offs_t offset, uint8_t data)
+{
+	m_treg3 = data;
+}
+
+uint8_t tmp94c241_device::t23mod_r()
+{
+	return m_t23mod;
+}
+
+void tmp94c241_device::t23mod_w(uint8_t data)
+{
+	m_t23mod = data;
+}
+
+uint8_t tmp94c241_device::trdc_r()
+{
+	return m_trdc;
+}
+
+void tmp94c241_device::trdc_w(uint8_t data)
+{
+	m_trdc = data;
+}
+
+void tmp94c241_device::treg4_w(offs_t offset, uint16_t data)
+{
+	m_treg4 = data;
+}
+
+void tmp94c241_device::treg5_w(offs_t offset, uint16_t data)
+{
+	m_treg5 = data;
+}
+
+void tmp94c241_device::treg6_w(offs_t offset, uint16_t data)
+{
+	m_treg6 = data;
+}
+
+void tmp94c241_device::treg7_w(offs_t offset, uint16_t data)
+{
+	m_treg7 = data;
+}
+
+void tmp94c241_device::treg8_w(offs_t offset, uint16_t data)
+{
+	m_treg8 = data;
+}
+
+void tmp94c241_device::treg9_w(offs_t offset, uint16_t data)
+{
+	m_treg9 = data;
+}
+
+void tmp94c241_device::trega_w(offs_t offset, uint16_t data)
+{
+	m_trega = data;
+}
+
+void tmp94c241_device::tregb_w(offs_t offset, uint16_t data)
+{
+	m_tregb = data;
+}
+
+uint8_t tmp94c241_device::t4mod_r()
+{
+	return m_t4mod;
+}
+
+void tmp94c241_device::t4mod_w(uint8_t data)
+{
+	m_t4mod = data;
+}
+
+uint8_t tmp94c241_device::t6mod_r()
+{
+	return m_t6mod;
+}
+
+void tmp94c241_device::t6mod_w(uint8_t data)
+{
+	m_t6mod = data;
+}
+
+uint8_t tmp94c241_device::t8mod_r()
+{
+	return m_t8mod;
+}
+
+void tmp94c241_device::t8mod_w(uint8_t data)
+{
+	m_t8mod = data;
+}
+
+uint8_t tmp94c241_device::tamod_r()
+{
+	return m_tamod;
+}
+
+void tmp94c241_device::tamod_w(uint8_t data)
+{
+	m_tamod = data;
+}
+
+uint8_t tmp94c241_device::t4ffcr_r()
+{
+	return m_t4ffcr;
+}
+
+void tmp94c241_device::t4ffcr_w(uint8_t data)
+{
+	change_timer_flipflop( 4, data & 3 );
+	change_timer_flipflop( 5, (data >> 6) & 3 );
+	m_t4ffcr = data | 0xc3;
+}
+
+uint8_t tmp94c241_device::t6ffcr_r()
+{
+	return m_t6ffcr;
+}
+
+void tmp94c241_device::t6ffcr_w(uint8_t data)
+{
+	change_timer_flipflop( 6, data & 3 );
+	change_timer_flipflop( 7, (data >> 6) & 3 );
+	m_t6ffcr = data | 0xc3;
+}
+
+uint8_t tmp94c241_device::t8ffcr_r()
+{
+	return m_t8ffcr;
+}
+
+void tmp94c241_device::t8ffcr_w(uint8_t data)
+{
+	change_timer_flipflop( 8, data & 3 );
+	change_timer_flipflop( 9, (data >> 6) & 3 );
+	m_t8ffcr = data | 0xc3;
+}
+
+uint8_t tmp94c241_device::taffcr_r()
+{
+	return m_taffcr;
+}
+
+void tmp94c241_device::taffcr_w(uint8_t data)
+{
+	change_timer_flipflop( 0xa, data & 3 );
+	change_timer_flipflop( 0xb, (data >> 6) & 3 );
+	m_taffcr = data | 0xc3;
+}
+
+uint8_t tmp94c241_device::t16run_r()
+{
+	return m_t16run;
+}
+
+void tmp94c241_device::t16run_w(uint8_t data)
+{
+	m_t16run = data;
+	for ( int i = 0; i < 4; i++ )
+	{
+		if ( !BIT(m_t16run, i) ) /* Timer isn't running */
+		{
+			/*
+			    These correspond to UP_COUNTER and TIMER_CHANGE
+			    for 16-bit timers 4, 6, 8 and A
+			*/
+			m_timer16[i] = 0;
+			m_timer_change[4 + i] = 0;
+		}
+	}
+}
+
+
+uint8_t tmp94c241_device::cap45_r(offs_t offset)
```

Comment:
Similarly to write, you can do:
uint16_t tmp94c241_device::cap45_r(offs_t offset)
{
    return m_t16_cap[offset+0];
}

or even a:

template<int timer> uint16_r cap_r()
{
   return m_t16_cap[timer];
}

and change the address map accordingly.